ci/scripts/install_python.sh

```diff
@@ -69,8 +68,32 @@ if [ $platform = "macOS" ]; then
     python="/Library/Frameworks/Python.framework/Versions/${version}/bin/python${version}"
     pip="${python} -m pip"
 
+    $python -m ensurepip
+    $pip install -U pip setuptools
+elif [ $platform = "Linux" ] ; then
+    echo "Downloading Python installer..."
+
+    if [ "$version" = "3.13" ];
+    then
+        fname="Python-${full_version}rc1.tar.xz"
+    else
+        fname="Python-${full_version}.tar.xz"
+    fi
+    wget "https://www.python.org/ftp/python/${full_version}/${fname}"
+
+    echo "Building and installing Python..."
```

Comment:
Let's not do that? Most people won't use a self-compiled Python, so it does not make much sense to test against that.

You can find backports for Ubuntu here: https://launchpad.net/~deadsnakes/+archive/ubuntu/ppa

Reply:
Or at worse you can use https://gregoryszorc.com/docs/python-build-standalone/main/

Reply:
At this point I think we might just drop testing on Ubuntu 20.04 for pyarrow and avoid issues (Ubuntu 20.04 is EOL in April 2025)